src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -52,570 +48,765 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
+    /*******************************************************************************************************
+    * @description Contact's owner has changed. Consider updating related account owner.
+    * @param contactOwnerChange a map of Contact Id to Contact, for the contacts to consider.
+    * @param dmlWrapper to hold the Contacts that need update.
+    */
+    private void updateOwners(Map<Id,Contact> contactOwnerChange, DmlWrapper dmlWrapper) {
+        List<Account> accountsToUpdate = new List<Account>();
+        Set<Id> contactsAccountId = new Set<Id>();
+
+        if (contactOwnerChange.size() == 0) {
+            return;
+        }
+
+        for (Contact con : contactOwnerChange.values()) {
+            contactsAccountId.add(con.AccountId);
+        }
+
+        List<Account> listAcc = [SELECT Id, RecordTypeId, Primary_Contact__c, OwnerId,
+                                (SELECT Id FROM Contacts limit 2)
+                                 FROM Account
+                                 WHERE Id IN :contactsAccountId];
+
+        for (Account acc : listAcc) {
+            //TODO: use setting to determine if Account owner should be changed (W-009780)
+            if (acc.Contacts.size() == 1 && acc.Primary_Contact__c != NULL &&
+                acc.RecordTypeId != NULL && (acc.RecordTypeId == userDefinedHHRecordTypeId
+                || acc.RecordTypeId == userDefinedAdminRecordTypeId))
+            {
+                Contact c = contactOwnerChange.get(acc.Primary_Contact__c);
+                Id ownerIdContact = (c != NULL ? c.OwnerId : NULL);
+
+                if (acc.OwnerId != ownerIdContact) {
+                    acc.OwnerId = ownerIdContact;
+                    accountsToUpdate.add(acc);
                 }
+
+             }
+        }
+
+        if (accountsToUpdate.size() > 0) {
+            dmlWrapper.objectsToUpdate.addAll((List<SObject>)accountsToUpdate);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Deletes the accounts for contacts that have been deleted if they are empty (no ops, no hh members).
+    * @param contactAccountDelete the list of Contacts that need to be considered.
+    * @param dmlWrapper to hold the Accounts that needs update.
+    */
+    public void deleteContactAccountsIfEmpty(List<Contact> contactAccountDelete, DmlWrapper dmlWrapper) {
+        List<Id> accountIds = new List<Id>();
+        List<Id> accountIdsUpdate = new list<Id>();
+        List<Account> accountsOnContacts = new List<Account>();
+        List<SObject> accountsForDeletion = new List<SObject>();
+
+        if (contactAccountDelete.size() == 0) {
+            return;
+        }
+
+        for (Contact thisContact : contactAccountDelete) {
+            accountIds.add(thisContact.accountId);
+        }
+
+        accountsOnContacts = [SELECT Id, Type, RecordTypeId,
+                            (SELECT Id FROM Opportunities LIMIT 1),
```

Comment:
Are we doing anything with Opportunties? (It's possible this is a GEM thing so perhaps we won't want to touch it, but I am going to tag Ben D or Luke P in an external comment here just in case for input. (cc: @lparrott or @bdvorachek -- are y'all using / needing this Opportunity subquery for GEM?)

Reply:
@lifewithryan  I'm not aware of anything, but line 779 here is where that sub-select gets used.

Reply:
aahhh thanks ben, I skimmed right past that apparently. :)